### PR TITLE
Support PINMemoryCache cost methods in PINCache

### PR DESCRIPTION
**Pull request by msqr**
This PR addresses issue #64 to expose methods on `PINCache` that accept a memory cache cost argument.

### CODE REVIEW

File: PINCache/PINCache.h [hunk 1]
```diff
@@ -165,6 +165,19 @@ typedef void (^PINCacheObjectContainmentBlock)(BOOL containsObject);
...
+ to go over the <memoryCache.costLimit> the cache is trimmed (oldest objects first). This method returns immediately
+ and executes the passed block after the object has been stored, potentially in parallel with other blocks
+ on the <concurrentQueue>.
+ 
+ @param object An object to store in the cache.
+ @param key A key to associate with the object. This string will be copied.
+ @param cost An amount to add to the <memoryCache.totalCost>.
+ @param block A block to be executed concurrently after the object has been stored, or nil.
+ */
+- (void)setObject:(id)object forKey:(NSString *)key withCost:(NSUInteger)cost block:(nullable PINCacheObjectBlock)block;
```
Comment (garrettmoon) [inline PINCache/PINCache.h]: Shouldn't this be:
`- (void)setObject:(id <NSCoding>)object forKey:(NSString *)key withCost:(NSUInteger)cost block:(nullable PINCacheObjectBlock)block;` ? (adding the NSCoding protocol to object)

File: PINCache/PINCache.h [hunk 2]
```diff
@@ -223,6 +236,17 @@ typedef void (^PINCacheObjectContainmentBlock)(BOOL containsObject);
...
 /**
+ Stores an object in the cache for the specified key and the specified memory cost. If the cost causes the total
+ to go over the <memoryCache.costLimit> the cache is trimmed (oldest objects first). This method blocks the calling thread
+ until the object has been stored.
+ 
+ @param object An object to store in the cache.
+ @param key A key to associate with the object. This string will be copied.
+ @param cost An amount to add to the <memoryCache.totalCost>.
+ */
+- (void)setObject:(nullable id)object forKey:(nullable NSString *)key withCost:(NSUInteger)cost;
```
Comment (garrettmoon) [inline PINCache/PINCache.h]: I think this should be:
`- (void)setObject:(id <NSCoding>)object forKey:(NSString *)key withCost:(NSUInteger)cost;` (id shouldn't be nullable should be conforming to NSCoding and key shouldn't be nullable)

**Comment by garrettmoon** [comment]
Thank you for putting this up, much appreciated! Also appreciate new tests :)

File: PINCache/PINCache.h (hunk 1 above)
Comment (msqr) [inline PINCache/PINCache.h]: Yes! This was a copy/paste from `PINMemoryCache`. I've pushed a correction now.

**Comment by garrettmoon** [comment]
Awesome, sorry this took so long to merge, really appreciate it!

**Pull request merged**